torchdata/datapipes/iter/util/distributed.py

```diff
@@ -93,7 +94,7 @@ def _done_callback_fn(self, index: int, f: Future):
         if f.exception():
             with self._lock:
                 self._end_flag = True
-        if self.callback_fn is not None:
+        if self.callback_fn is not None and not self._is_shutdown:
```

Comment:
Without this, it may attempts to submit the `callback_fn` even when the `_executor` has been shut down.

cc: @ejguan

Reply:
It seems reasonable.
Then, as the dev note, can you pls add some comments about the difference between `end_flag` and `is_shutdown`? Like `end_flag` indicates the end of epoch or exception,  and callback function will be executed to handle the exception.